### PR TITLE
feat: support EXPLAIN in mysql

### PR DESCRIPTION
**Pull request by xiaoyu-meng-mxy**
Currently EXPLAIN is not supported in sqlglot.dialects.mysql. 

Because 'The [DESCRIBE](https://dev.mysql.com/doc/refman/8.4/en/describe.html) and [EXPLAIN](https://dev.mysql.com/doc/refman/8.4/en/explain.html) statements are synonyms.', this pr supports EXPLAIN by treat it as DESCRIBE

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -2567,7 +2567,10 @@ def _parse_returns(self) -> exp.ReturnsProperty:
 
     def _parse_describe(self) -> exp.Describe:
         kind = self._match_set(self.CREATABLES) and self._prev.text
-        style = self._match_texts(("EXTENDED", "FORMATTED", "HISTORY")) and self._prev.text.upper()
+        style = (
+            self._match_texts(("ANALYZE", "EXTENDED", "FORMATTED", "HISTORY"))
```
Comment (georgesittas) [inline sqlglot/parser.py:2571]: Let's make this a class-level set constant and reference that instead.

File: tests/dialects/test_mysql.py [hunk 2]
```diff
@@ -1305,3 +1305,12 @@ def test_grant(self):
         for sql in grant_cmds:
             with self.subTest(f"Testing MySQL's GRANT command statement: {sql}"):
                 self.validate_identity(sql, check_command_warning=True)
+
+    def test_explain(self):
+        self.validate_identity(
+            "EXPLAIN ANALYZE SELECT * FROM t", write_sql="DESCRIBE ANALYZE SELECT * FROM t"
```
Comment (georgesittas) [inline tests/dialects/test_mysql.py:1311]: [Nit]: we don't usually specify `write_sql` as a kwarg.
```suggestion
            "EXPLAIN ANALYZE SELECT * FROM t",
            "DESCRIBE ANALYZE SELECT * FROM t",
```

**Review by georgesittas — APPROVED** [review]
LGTM, thanks for the PR. Just a couple minor comments.

File: sqlglot/parser.py [hunk 3]
```diff
@@ -1286,6 +1286,8 @@ class Parser(metaclass=_Parser):
 
     PRIVILEGE_FOLLOW_TOKENS = {TokenType.ON, TokenType.COMMA, TokenType.L_PAREN}
 
+    DESCRIBE_STATEMENT_STYLES = {"ANALYZE", "EXTENDED", "FORMATTED", "HISTORY"}
```
Comment (xiaoyu-meng-mxy) [inline sqlglot/parser.py:1289]: I see "HISTORY" is added for dialects.databricks

I can also remove HISTORY from here and override it in databricks parser, if necessary. And do the same for ANALYZE in mysql parser. Not sure whether this is needed 🙏

File: sqlglot/parser.py (hunk 1 above)
Comment (xiaoyu-meng-mxy) [reply sqlglot/parser.py]: Makes sense. Thanks for the quick review!

File: sqlglot/parser.py (hunk 3 above)
Comment (barakalon) [reply sqlglot/parser.py:1289]: sqlglot values brevity.

How about just "DESCRIBE_STYLES"?

**Review by barakalon — APPROVED** [review]

**Review by georgesittas — APPROVED** [review]
Thank you for the PR!

**Pull request merged**